### PR TITLE
Added setting to allow requires/imports to be added within the current comment form during code action and completion. [#1316]

### PR DESCRIPTION
**Pull request by blnote**
- [x] I added a new entry to [CHANGELOG.md](https://github.com/clojure-lsp/clojure-lsp/blob/master/CHANGELOG.md)
- [x] I updated documentation if applicable (`docs` folder)

Added setting to allow requires and imports to be added within the current comment form during code action and completion. [#1316](https://github.com/clojure-lsp/clojure-lsp/issues/1316)

 `:completion {:add-libs-to-rcf [true|false|nil]}`

- `false`: everything works as before
- `nil` or missing: when a code action for adding a missing lib is executed within a comment form, the user will be asked where the form shall be included
- `true`:  `imports` and `requires` from within a comment form are always added to the top of the form.
 
### Discussion

I like the enhancement, because it makes it possible to eval the require without scrolling to the top of the namespace. However,  this is just a suggestion :)
I started with a simpler refactoring separately form the existing code, but ended up duplicating the pattern-matching logic of `add-missing-libspec/add-to-namespace*`, so I extracted the shared part into two separate functions. (did not change the logic, but lots of added lines because of it) Certainly open for discussion.

### Notes on code changes
- added `edit/inside-rcf?` to test if we are in comments
- `add-missing-libspec/add-to-namespace*` was already quite long and the comment-case differed in a few points (e.g. modifying the comment form while the cursor is inside caused conflicts with completion when the whole edited form is returned) so I decided to create a separate function `add-to-rcf*`. which of the two function is chosen depends on whether the zloc-argument points to a comment form.
- since cursor position had been lost in completions with delayed resolving, I modified `completion.clj`: the position of the comment form is extracted in `completion-item-with-unresolved-alias-edit` and passed along as `rcf-pos`, the comment form is then recovered in `completion-item-with-alias-edit` for resolving

### CODE REVIEW

File: lib/src/clojure_lsp/feature/add_missing_libspec.clj [hunk 1]
```diff
@@ -175,61 +224,112 @@
...
+    (when (and
+            (not (= false? c-setting))
+            rcf-zloc
+            (or c-setting
+                (and
+                  producer
+                  (= "Yes"
+                     (producer/show-message-request
+                       producer
+                       "Add require inside this comment form? (to avoid this prompt, set { :completion {:add-libs-inside-rcf true|false }.)"
```
Comment (ericdallo) [inline lib/src/clojure_lsp/feature/add_missing_libspec.clj:257]: I'd leave this as:
```suggestion
                       "Add require inside this comment form?"
```

And if anyone complains we can point to the docs

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@
 - General
   - Bump clj-kondo to `2023.12.16-20240202.114914-15`.
 
+- Editor
+  - Added setting to allow requires and imports to be added within the current comment form during code action and completion: `:completion {:add-libs-to-rcf [true|false|nil]}` #1316
```
Comment (ericdallo) [inline CHANGELOG.md:9]: Please remember to update settings.md + docs/all-available-settings.md with the default value

File: lib/src/clojure_lsp/feature/add_missing_libspec.clj (hunk 1 above)
Comment (blnote) [reply lib/src/clojure_lsp/feature/add_missing_libspec.clj:257]: thanks for your quick response - sure, that's fine with me, I'll update code & docs (forgot to push them yesterday...)
I wrote the prompt before adding the functionality for imports, maybe we should ask "Add lib inside this comment form?" instead?

Comment (blnote) [reply lib/src/clojure_lsp/feature/add_missing_libspec.clj:257]: Nevermind, I can just parametrize it and ask for require and import respectively.

**Comment by blnote** [comment]
While testing I thought I created a problem, but turned out it was issue [1754](https://github.com/clojure-lsp/clojure-lsp/issues/1754). The problem seems to be in `add_missing_libspec/find-when-starts-with-sym` .
A quick fix could be to not only look if the existing libs **starts with** the lib-to-add, but also check for **equality**  whenever the existing lib-import is in a list/vector.
this fixes the test-case and all tests run through, but I'm not sure... could be that to make it more robust, one would have to look at **all** occurrences of strings in ns of which the lib-to-add is a prefix and choose a suitable one. Don't know if you worked on this already?

**Comment by ericdallo** [comment]
@blnote thanks, looks good, I'll do some manual tests, could you do a one last solve conflict please?

Also, I intend to fix #1754 soon

**Review by ericdallo — APPROVED** [review]

**Review by ericdallo — APPROVED** [review]
Thank you!

**Pull request merged**